src/main/java/com/internet/shop/services/OrderServiceImpl.java

```diff
@@ -0,0 +1,50 @@
+package com.internet.shop.services;
+
+import com.internet.shop.dao.OrderDao;
+import com.internet.shop.lib.Inject;
+import com.internet.shop.lib.Service;
+import com.internet.shop.models.Order;
+import com.internet.shop.models.ShoppingCart;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class OrderServiceImpl implements OrderServiceInterface {
+
+    @Inject
+    OrderDao orderDao;
+
+    @Inject
+    ShoppingCartServiceIntercace shoppingCartService;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        Order order = new Order();
+        order.setProducts(List.copyOf(shoppingCart.getProducts()));
+        orderDao.create(order);
+        shoppingCartService.clear(shoppingCart);
+        return order;
+    }
+
+    @Override
+    public List<Order> getUserOrders(Long userId) {
+        return getAll().stream()
+                .filter(order -> order.getId().equals(userId))
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public Order get(Long id) {
+        return orderDao.get(id).orElseThrow();
```

Comment:
According to Sofia's review, there is no need to throw an exception^ because in case of element being null you'll get an exception while extracting the value.

Reply:
fixed